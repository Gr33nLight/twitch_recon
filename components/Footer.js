import React from 'react';
import { Box, Center, Icon, Link, useColorMode } from '@chakra-ui/react';
import { FaTwitter, FaReddit } from 'react-icons/fa';

const Footer = () => {
  const { colorMode, toggleColorMode } = useColorMode();

  return (
    <Box flexShrink={0}>
      <Center>
        <Icon
          as={FaTwitter}
          w={5}
          h={5}
          mx={4}
          color={colorMode == 'light' ? 'gray.600' : 'purple.400'}
          cursor="pointer"
          onClick={() =>
            window.open(
              'http://twitter.com/share?text=Hey! Check out Twitch Recon! It lets you sync twitch vods across various channels&url=https://twitchrecon.com&hashtags=twitchrecon',
              '_blank'
            )
          }
        />
        <Icon
          as={FaReddit}
          w={5}
          h={5}
          mx={4}
          color={colorMode == 'light' ? 'gray.600' : 'purple.400'}
          cursor="pointer"
          onClick={() =>
            window.open(
              'http://www.reddit.com/submit?url=https://twitchrecon.com/&title=Check%20out%20twitchrecon!',
              '_blank'
            )
          }
        />
      </Center>
      <Center ml="10px" mt="10px" fontSize="11px" float="left">
        Made with ❤️ by{' '}
        <Link
          color={colorMode == 'light' ? 'gray.600' : 'purple.400'}
          href="https://twitter.com/Gr33nLight_"
        >
          Gr33nLight
        </Link>
      </Center>
    </Box>
  );
};

export default Footer;
