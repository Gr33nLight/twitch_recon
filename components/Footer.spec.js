import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';

describe('<Footer />', () => {
  it('renders correctly', async () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
