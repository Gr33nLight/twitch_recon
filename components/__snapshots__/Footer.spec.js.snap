// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<div
  className="css-fp7pz9"
>
  <div
    className="css-gmuwbf"
  >
    <svg
      className="chakra-icon css-1f59v40"
      fill="currentColor"
      focusable={false}
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      />
    </svg>
    <svg
      className="chakra-icon css-1f59v40"
      fill="currentColor"
      focusable={false}
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M201.5 305.5c-13.8 0-24.9-11.1-24.9-24.6 0-13.8 11.1-24.9 24.9-24.9 13.6 0 24.6 11.1 24.6 24.9 0 13.6-11.1 24.6-24.6 24.6zM504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-132.3-41.2c-9.4 0-17.7 3.9-23.8 10-22.4-15.5-52.6-25.5-86.1-26.6l17.4-78.3 55.4 12.5c0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.3 24.9-24.9s-11.1-24.9-24.9-24.9c-9.7 0-18 5.8-22.1 13.8l-61.2-13.6c-3-.8-6.1 1.4-6.9 4.4l-19.1 86.4c-33.2 1.4-63.1 11.3-85.5 26.8-6.1-6.4-14.7-10.2-24.1-10.2-34.9 0-46.3 46.9-14.4 62.8-1.1 5-1.7 10.2-1.7 15.5 0 52.6 59.2 95.2 132 95.2 73.1 0 132.3-42.6 132.3-95.2 0-5.3-.6-10.8-1.9-15.8 31.3-16 19.8-62.5-14.9-62.5zM302.8 331c-18.2 18.2-76.1 17.9-93.6 0-2.2-2.2-6.1-2.2-8.3 0-2.5 2.5-2.5 6.4 0 8.6 22.8 22.8 87.3 22.8 110.2 0 2.5-2.2 2.5-6.1 0-8.6-2.2-2.2-6.1-2.2-8.3 0zm7.7-75c-13.6 0-24.6 11.1-24.6 24.9 0 13.6 11.1 24.6 24.6 24.6 13.8 0 24.9-11.1 24.9-24.6 0-13.8-11-24.9-24.9-24.9z"
      />
    </svg>
  </div>
  <div
    className="css-17woz96"
  >
    Made with ❤️ by
     
    <a
      className="chakra-link css-irhq0v"
      href="https://twitter.com/Gr33nLight_"
    >
      Gr33nLight
    </a>
  </div>
</div>
`;
